core/commands/refs.go

```diff
@@ -86,6 +89,12 @@ Note: List all referencess recursively by using the flag '-r'.
 			return
 		}
 
+		cached, _, err := req.Option("cached").Bool()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
```

Comment:
@whyrusleeping I am not sure how can I do without the `Cached` logic. The `NodeGetter.Get` hangs when I use with BlockService with offline exchange and the ref is not cached locally. Looks like I've to change quite a lot if I want to make nodePromise(i.e. NodeGetter) to respond even if the ref is not locally available.